lib/maths/CBoostedTreeImpl.cc

```diff
@@ -466,9 +469,13 @@ CBoostedTreeImpl::~CBoostedTreeImpl() = default;
 CBoostedTreeImpl& CBoostedTreeImpl::operator=(CBoostedTreeImpl&&) = default;
 
 void CBoostedTreeImpl::train(core::CDataFrame& frame,
-                             const TProgressCallback& recordProgress,
-                             const TMemoryUsageCallback& recordMemoryUsage,
                              const TTrainingStateCallback& recordTrainStateCallback) {
+    if (m_Instrumentation == nullptr) {
+        HANDLE_FATAL(<< "Internal error: analysis state was not initialize. Please report this problem.");
```

Comment:
This is causing unit test failures. I'm not sure we should enforce this: it doesn't feel fatal that the process isn't monitored: it's up to the caller. I think I'd rather just set `recordProgress` and `recordMemoryUsage` to stubs if this is null.

Reply:
Thank you for catching this., I solved the issue.